terraform/vals.tf

```diff
@@ -0,0 +1,24 @@
+# Copyright 2020 Google LLC
+# Copyright 2020 CriticalBlue Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+locals {
```

Comment:
I really think this is too much magic and too many layers of indirection.

Reply:
OK - I've stripped it back to the simplest it could be while keeping the flexibility. Everything is now exposed and only the comments indicate the dependency between appengine_location and appengine_region. I updated the README to reflect the change.